src/main/webapp/app/overview/course-score-calculation.service.ts

```diff
@@ -26,20 +28,39 @@ export class CourseScoreCalculationService {
     calculateTotalScores(courseExercises: Exercise[], course: Course): Map<string, number> {
         const scores = new Map<string, number>();
         let pointsAchievedByStudentInCourse = 0.0;
-        let maxPointsInCourse = 0;
-        let reachableMaxPointsInCourse = 0;
+        let maxPointsInCourse = 0; // the sum of all included exercises, whose due date is over or unset or who are automatically assessed and assessment is done
+        let reachableMaxPointsInCourse = 0; // the sum of all included and already assessed exercises
         let presentationScore = 0;
         for (const exercise of courseExercises) {
             const isExerciseFinished = !exercise.dueDate || exercise.dueDate.isBefore(dayjs());
             const isAssessmentOver = !exercise.assessmentDueDate || exercise.assessmentDueDate.isBefore(dayjs());
             const isExerciseIncluded = exercise.includedInOverallScore !== IncludedInOverallScore.NOT_INCLUDED;
-
-            if (isExerciseFinished && isExerciseIncluded) {
+            const isExerciseAssessedAutomatically = exercise.type === ExerciseType.PROGRAMMING && exercise.assessmentType === AssessmentType.AUTOMATIC;
+            const isAutomaticAssessmentFinished =
+                isExerciseAssessedAutomatically &&
+                (!(exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate ||
+                    (exercise as ProgrammingExercise).buildAndTestStudentSubmissionsAfterDueDate!.isBefore(dayjs()));
+            /*
+            The requirement that has to be fulfilled for every exercise: It has to be included in score
+            The base case we include: An exercise that is not an automatically assessed programming exercise and whose due date is over
+            Edge case 1: An automatically assessed programming exercise without test runs after the due date
+                -> include in maxPointsInCourse directly after release because the student can achieve points immediately
+            Edge case 2: An automatically assessed programming exercise with test runs after the due date
+                -> include in maxPointsInScore after the final test run is over, not immediately after release because
+                   the test run after due date is important for the final course score (hidden tests)
+             */
+            if ((isAutomaticAssessmentFinished || (!isExerciseAssessedAutomatically && isExerciseFinished)) && isExerciseIncluded) {
```

Comment:
```suggestion
            if (isExerciseIncluded && ((!isExerciseAssessedAutomatically && isExerciseFinished) || isAutomaticAssessmentFinished)) {
```
Reordering them to follow the explanation in the comment makes the condition more understandable.